module5/lessons/Week4/CallingThirdPartyAPIs.md

```diff
@@ -0,0 +1,97 @@
+
+Today we're going to take the exciting leap of calling APIs, such as those listed above, from your applications! You will learn how to retrieve data from the API and then display it to the user.
+
+
+### Preparation Review
+
+In your preparation you deepened your understanding of asynchronous code. 
+
+<section class="call-to-action" markdown="1">
+With your partner: Discuss your current understanding of asynchronous code and the keywords `Async` and `Await`. When we come back together you instructor will call on folks to share what they talked out.
+</section>
+
+<section class="call-to-action" markdown="1">
+With your partner: Discuss why we might want to use asynchronous code when calling a third party API from our applications? When we come back together you instructor will call on folks to share what they talked out.
+</section>
+
+<section class="answer" markdown="1">
+### Asynchronous Code Definitions
+
+**Synchronous Code**: Synchronous code executes tasks sequentially, waiting for each to finish before proceeding, potentially causing resource inefficiencies during idle periods.
+
+**Asynchronous Code**: Asynchronous code enables tasks to execute independently, improving resource utilization by allowing operations to progress without waiting for others to complete.
+
+**Async Keyword**: In C#, the async keyword is used to mark a method as asynchronous. It indicates that the method contains asynchronous operations and may include one or more await expressions. Asynchronous methods can execute tasks concurrently without causing blocking.
+
+**Await Keyword**: The await keyword is used within an asynchronous method to pause execution and asynchronously wait for the completion of an asynchronous operation. When await is used, it allows the method to temporarily yield control until the awaited task is finished executing. This non-blocking behavior enables efficient resource usage and helps maintain the application's responsiveness.
+
+</section>
+
+
+### Learning From a Tutorial
+Follow [this](https://www.ezzylearning.net/tutorial/how-to-consume-third-party-web-apis-in-asp-net-core) tutorial individually up until the section **Managing HttpClient objects with IHttpClientFactory**. You will have roughly 1.5 hours, which should be enough time to thoughtfully work through it.
+
+As you work through the tutorial, add comments to specify what each part of your code is doing. I also highly recommend that you type the code instead of copy/pasting for topics that feel new.
+
+As with all tutorials, you have to expect it might not work perfectly and you might need to do some debugging to get your code to work! If you're stuck on something for more than 5 or 10 minutes reach out to your peers in code-help!
+
+Deliverable✅: Once you have finished the tutorial, send a message to your instructor with your answer to the following question:

```

Comment:
I wonder if we should also have a deliverable question that is related to the tutorial outcome? Something to put some pressure on to dig in and get through the tutorial fully

Reply:
Ope - now I see the other questions below :)

Reply:
I waws thinking about this too. I wish the students were all able to self motivate at this point and keep pushing forward through the deliverable, but might be best to have a "did you do the thing?" part of the deliverable too. Maybe a screenshot of searching for a country outside of the US.